fix: use nix signal feature to allow for dep upgrade

hseeberger opened the pull request:
Ref #5936.

dependabot[bot] committed:
build: bump nix from 0.26.2 to 0.27.1

Bumps [nix](https://github.com/nix-rust/nix) from 0.26.2 to 0.27.1.
- [Changelog](https://github.com/nix-rust/nix/blob/master/CHANGELOG.md)
- [Commits](https://github.com/nix-rust/nix/compare/v0.26.2...v0.27.1)

---
updated-dependencies:
- dependency-name: nix
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

hseeberger committed:
fix: use nix signal feature to allow for dep upgrade

mrinalwadhwa commented:
@hseeberger awesome! Thank you, while we review please

- [ ] Accept the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding your Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

The easiest way to do this is to [edit the CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/edit/main/CONTRIBUTORS.csv) file in the github web UI and create a PR, this will make the commit as verified.

mrinalwadhwa commented:
Nice the change works! There's a minor formatting/spacing error, if you could address that and accept the CLA, we can merge 👍  https://github.com/build-trust/ockam/actions/runs/6189600055/job/16804078484?pr=5974#step:5:610

metaclips committed:
ci: use nix to run ockam command test

metaclips committed:
ci: use nix env for rust doc test

metaclips commented on implementations/rust/ockam/ockam_api/Cargo.toml:
```diff
@@ -43,7 +43,7 @@ kafka-protocol = "0.6.1"
 miette = "5.10.0"
 minicbor = { version = "0.19.0", features = ["alloc", "derive"] }
-nix = "0.27"
+nix = {verion = "0.27", features = ["signal"] }
```
There's a typo here, `verion`.

hseeberger replied:
Bummer!

hseeberger replied:
Should be fixed now 🤞

hseeberger replied:
I built locally before and it (seemed to) work(ed). Not sure what happened. There are lots of crates, though, and some might not be built when just using `cargo check` form the workspace directory. Anyway, this time I actually built that particular crate, and even tested it: `cargo test -p ockam_api`

metaclips reviewed (changes requested):
Little changes and accepting the CLA, and we should merge. Thanks @hseeberger

dependabot[bot] committed:
build: bump @types/node in /implementations/typescript

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 20.6.0 to 20.6.1.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
ci: bump Swatinem/rust-cache from 2.6.2 to 2.7.0

Bumps [Swatinem/rust-cache](https://github.com/swatinem/rust-cache) from 2.6.2 to 2.7.0.
- [Release notes](https://github.com/swatinem/rust-cache/releases)
- [Changelog](https://github.com/Swatinem/rust-cache/blob/master/CHANGELOG.md)
- [Commits](https://github.com/swatinem/rust-cache/compare/v2.6.2...a95ba195448af2da9b00fb742d14ffaaf3c21f43)

---
updated-dependencies:
- dependency-name: Swatinem/rust-cache
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

adrianbenavides committed:
fix(rust): resize icons png files to 24x24 so they don't look blurry on `MacOS`

adrianbenavides committed:
feat(rust): persist tcp inlets data to reuse the same address when recreating them

adrianbenavides committed:
feat(rust): add menu items to manually connect/disconnect to an invited service

hseeberger commented:
Accepted the CLA: https://github.com/build-trust/ockam-contributors/pull/274

hseeberger commented:
Will close this PR and create a new one against develop!

hseeberger commented:
See #6006.
